addons/sale_timesheet/models/project.py

```diff
@@ -569,6 +569,13 @@ def _compute_sale_line(self):
             if task.allow_billable and not task.sale_line_id:
                 task.sale_line_id = task._get_last_sol_of_customer()
 
+    @api.onchange("partner_id")
+    def onchange_partner_id(self):
+        super().onchange_partner_id()
+        if self.allow_billable and not self.sale_line_id:
+                self.sale_line_id = self._get_last_sol_of_customer()
+
+
```

Comment:
visually, i think you have indentation problem, remove also the extra blank line
```suggestion
        super().onchange_partner_id()
        if self.allow_billable and not self.sale_line_id:
            self.sale_line_id = self._get_last_sol_of_customer()

```